### DISCUSSION TITLE
What are the advantages of using external speaker embeddings over internal ones ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#2060] [General Q&A] [answered]
Hi,

I am training a multi speaker VITS model and I came across two (exclusive) options : 
- Using `use_speaker_embedding=True` and training goes smoothly
- Using `use_d_vector_file=True` as many people seem to prefer

I could not notice any performance (in training time) improvements using d_vector_file over the first option. 
Generated speech sounded as good in both cases so what are the advantages of d_vector_file ?

Thank you for the clarification

**Comment by p0p4k** [comment, marked as answer]
It just decouples the processes.

**Reply by Ca-ressemble-a-du-fake** [reply]
Ah OK thank you @p0p4k! Now I remember reading something along those lines : it is more convenient to have speaker embeddings as external files because in case of fine tuning you just have to provide the speaker embeddings for the new fine tuned voices whereas with internal embeddings the previous ones are built-in in the model and so have to be provided again. Now that I search in gitter I find [@erogol answer](https://gitter.im/coqui-ai/TTS?at=6303a046aa09177429a8a786) which adds some clarifications too : 
> if you want to keep the speakers from the previous run, yes you should keep the embeddings. It should not affect much for the new speakers as long as the new dataset is large/good enough

**Reply by p0p4k** [reply]
To add on top of it I would recommend anyone serious about TTS to read about at least the basic architectures and working of the models from the paper (maybe short summary, not detailed math). It never hurts to know more. Things will magically make sense if you do so. Good luck!
